cookbooks/vectortile/recipes/default.rb

```diff
@@ -0,0 +1,211 @@
+  database "tiles"
+  owner "tileupdate"
+  permissions "tileupdate" => :all, "tilekiln" => :usage
+end
+
+postgresql_database "spirit" do
+  cluster node[:vectortile][:database][:cluster]
+  owner "tileupdate"
+end
+
+postgresql_extension "postgis" do
+  cluster node[:vectortile][:database][:cluster]
+  database "spirit"
+end
+
+# Get a recent osm2pgsql version with backports.
+apt_preference "osm2pgsql" do
+  pin "release o=Debian Backports"
+  pin_priority "600"
+end
+
+apt_package "osm2pgsql"
+
+style_directory = "/srv/vector.openstreetmap.org/spirit"
+git style_directory do
+  repository "https://github.com/pnorman/spirit.git"
+  # Check out head for now
+  # revision "61db723"
+  user "tileupdate"
+  group "tileupdate"
+end
+
+themepark_directory = "/srv/vector.openstreetmap.org/osm2pgsql-themepark"
+git themepark_directory do
+  repository "https://github.com/osm2pgsql-dev/osm2pgsql-themepark.git"
+  user "tileupdate"
+  group "tileupdate"
+end
+
+tilekiln_directory = "/opt/tilekiln"
+
+python_virtualenv tilekiln_directory do
+  interpreter "/usr/bin/python3"
+end
+
+python_package "tilekiln" do
+  python_virtualenv tilekiln_directory
+  python_version "3"
+  version "0.5.1"
+end
+
+template "/srv/vector.openstreetmap.org/html/index.html" do
+  source "index.html.erb"
+  owner "www-data"
+  group "www-data"
+  mode "644"
+end
+
+directory "/srv/vector.openstreetmap.org/data" do
+  user "tileupdate"
+  group "tileupdate"
+  mode "755"
+end
+
+template "/usr/local/bin/import-planet" do
+  source "import-planet.erb"
+  owner "root"
+  group "root"
+  mode "755"
+end
+
+template "/usr/local/bin/tilekiln-storage-init" do
+  source "tilekiln-storage-init.erb"
+  owner "root"
+  group "root"
+  mode "755"
+  variables :tilekiln_bin => "#{tilekiln_directory}/bin/tilekiln", :storage_database => "tiles", :config_path => "#{style_directory}/spirit.yaml"
+end
+
+directory "/srv/vector.openstreetmap.org/complete" do
+  user "root"
+  group "root"
+  mode "755"
+end
+
+execute "tilekiln-storage-init" do
+  command "/usr/local/bin/tilekiln-storage-init"
+  user "tileupdate"
+  not_if { ::File.exist?("/srv/vector.openstreetmap.org/complete/tilekiln-storage-init") }
+  notifies :create, "file[/srv/vector.openstreetmap.org/complete/tilekiln-storage-init]", :immediately
+end
+
+file "/srv/vector.openstreetmap.org/complete/tilekiln-storage-init" do
+  action :nothing
+  content "lockfile for tilekiln-storage-init"
+end
+
+postgresql_table "metadata" do
+  cluster node[:vectortile][:database][:cluster]
+  database "tiles"
+  schema "tilekiln"
+  owner "tileupdate"
+  permissions "tileupdate" => :all, "tilekiln" => :select
+end
+
+systemd_service "tilekiln" do
+  description "Tilekiln vector tile server"
+  user "tilekiln"
+  after "postgresql.service"
+  wants "postgresql.service"
+
+  exec_start "#{tilekiln_directory}/bin/tilekiln serve static --storage-dbname tiles --num-threads #{node[:vectortile][:serve][:threads]}"
```

Comment:
Can we add `sandbox true` here, maybe with some additional relaxations? What does `tilekiln serve` need to be able to do/access?